sdk/test-utils/test-utils/test/tracing/instrumentorTracingSpanTest.spec.ts

```diff
@@ -0,0 +1,177 @@
+      // TODO: figure out how to be smarter about not wrapping sync functions in promise...
+      const result = await instrumenter.withContext(tracingContext, function() {
+        assert.equal(instrumenter.currentContext(), tracingContext);
+        return 42;
+      });
+
+      assert.equal(result, 42);
+      assert.notEqual(instrumenter.currentContext(), tracingContext);
+    });
+    it("sets the active context during async functions", async function() {
+      const { tracingContext } = instrumenter.startSpan("contextTest");
+      const result = await instrumenter.withContext(tracingContext, async function() {
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        assert.equal(instrumenter.currentContext(), tracingContext);
+        return 42;
+      });
+      assert.equal(result, 42);
+      assert.notEqual(instrumenter.currentContext(), tracingContext);
+    });
+    it("resets the previous context after the function returns", async function() {
+      const existingContext = instrumenter.currentContext();
+      const { tracingContext } = instrumenter.startSpan("test");
+      await instrumenter.withContext(tracingContext, async function() {
+        // no-op
+      });
+      assert.equal(instrumenter.currentContext(), existingContext);
+    });
+  });
+});
+
+describe("TestInstrumenter with MockClient", function() {
+  let instrumenter: TestInstrumenter;
+  let client: MockClientToTest;
+  beforeEach(function() {
+    instrumenter = new TestInstrumenter();
+    useInstrumenter(instrumenter);
+    client = new MockClientToTest();
+  });
+  it("starts a span and adds to startedSpans array", async function() {
+    await client.mockGetMethod();
+    assert.equal(instrumenter.startedSpans.length, 1);
+    assert.equal(instrumenter.startedSpans[0].name, "MockClientToTest.mockGetMethod");
+  });
+});
+
+describe("Test supportsTracing functionality", function() {
+  let client: MockClientToTest;
+  beforeEach(function() {
+    client = new MockClientToTest();
+  });
+  it("supportsTracing with the setMethod", function() {
+    assert.supportsTracing((options) => client.mockSetMethod({ key: "value" }, options), [
+      "MockClientToTest.mockSetMethod"
+    ]);
+    expect((options: any) => client.mockSetMethod({ key: "value" }, options)).to.supportsTracing([
+      "MockClientToTest.mockSetMethod"
+    ]);
+  });
+});
+// or something that has upgraded to core-tracing preview.14
+export class MockClientToTest {
+  private record: Record<string, string>;
+  tracingClient: TracingClient;
+
+  constructor() {
+    this.record = {};
+    this.tracingClient = createTracingClient({
+      namespace: "Microsoft.Test",
+      packageName: "@azure/test",
+      packageVersion: "foobar"
+    });
+  }
+  // const myOperationResult = await withSpan("myClassName.myOperationName", (updatedOptions) => myOperation(updatedOptions), options);
+  async mockSetMethod(record: any, options?: any) {
```

Comment:
I think all the tests just use one of these methods. Can we just rename this to `method(options?: { tracingOptions?: OperationTracingOptions }` and get rid of the `record` and the console.log statements?